content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,226 @@
+- `float` — e.g. `0.4`
+- `int` — e.g. `42`
+- `null` — `null`, `~`, and when value is omitted.
+- `string` — e.g. `""`, `"ConfigMap"`, `"0xbeadcafe"`
+- `any` — any valid YAML value is permitted. Must be set [explicitly](#schematype).
+- `map` — A YAML mapping that must contain exactly one item of each given key, whose value must be of the type specified.
+- `array` — A YAML sequence, each item must be of the specified `type`.
+
+---
+## Inferring Types
+
+Structure in Schema is largely expressed by example rather than by description. Types are — by default — inferred based on the values given. The Configuration Author can override these defaults using the [`@schema/type`](#schematype) annotation.
+
+```yaml
+#@schema/match data_values=True
+---
+system_domain: ""
+
+load_balancer:
+  enable: true
+  static_ip: ""
+
+app_domains:
+- ""
+
+databases:
+- name: ""
+  adapter: postgresql
+  host: ""
+  port: 5432
+  user: admin
+  secretRef:
+    name: ""
+```
+
+where:
+- Data Values must consist of four top-level map items:
+    - `system_domain` — whose value is a string,
+    - `load_balancer` — whose value is a map containing two items:
+        - `enable` — whose value is a boolean
+        - `static_ip` — whose value is a string
+    - `app_domains` — whose value is an array of strings (and only strings)
+    - `databases` — whose values is an array of maps of a specific shape: six (6) items:
+        - `name` — containing a string
+        - `adapter` — containing a string
+        - `host` — also holds a string
+        - `port` — holds an integer
+        - `user` — a string
+        - `secretRef` — containing a map with one key:
+            - `name` — containing a string
+
+## Inferring Default Values
+
+The exact values specified in Schema are the defaults (with one notable exception — arrays — which are [detailed below](#inferring-defaults-for-arrays)). This ensures that all nodes are initialized with values, with reasonable defaults.
+
+Configuration Consumers should specify in their data values file only those values which vary from the default, reducing risk of errors and generally making configuration easier to maintain.
+
+### Inferring Defaults for Scalars
+
+When a scalar value is specified, it is simply the default.  From the example, above:
+
+- `system_domain` is an empty string, by default.
+- `load_balancer.enable` is true by default.
+- `databases[].adapter` is `"postgres"`, by default.
+
+### Inferring Defaults for Mappings
+
+The set of items of a map are its default: any item missing will be automatically inserted with _its_ defaults.
+
+- if `load_balancer` is omitted from supplied Data Values, it is inserted:
+    - `load_balancer.enabled` is `true`
+    - `load_balancer.static_ip` is `""`
+- if `load_balancer` is specified with only `static_ip`, `enabled` is `true`.
+
+When the Configuration Consumer supplies an empty map (i.e. `{}`), all items are defaulted:
+
+For example,
+
+`load_balancer: {}`
```

Comment:
```suggestion
    ```yaml
    load_balancer: {} 
    ```
```
nit: it helps when the rendered bits are similarly formatted.

Reply:
oof